### PR TITLE
feat(oauth): add oauth success screen

### LINKED ISSUE #6996: Add an OAuth success screen at /oauth/success/<client_id>
Ref https://bugzilla.mozilla.org/show_bug.cgi?id=1528355

Both the reference browser and Fenix complete their flow by redirecting to `https://accounts.firefox.com/oauth/success/<client_id>`, which currently doesn't exist.

Let's show a placeholder similar to the Sync success screen so the browser
doesn't need to redirect *back* to `/connect_another_device?showSuccessMessage=true`.

cc @ryanfeeley

### PR DESCRIPTION
**Pull request by vladikoff**
Co-authored-by: Vlad Filippov <vlad.filippov@gmail.com>
Co-authored-by: hritvi <hritvi2801@gmail.com>

Fixes #6996

### CODE REVIEW

File: app/scripts/models/reliers/oauth.js [hunk 1]
```diff
@@ -171,13 +177,24 @@ var OAuthRelier = Relier.extend({
...
     // is used in the verification flow, it'll be set to the `client_id`.
     if (this.getSearchParam('service')) {
       throw OAuthErrors.toInvalidParameterError('service');
     }
   },
 
+  _setupSuccessFlow () {
+    const pathname = this.window.location.pathname.split('/');
+    const clientId = pathname[pathname.length - 1];
+    if (! clientId) {
```
Comment (shane-tomlinson) [inline app/scripts/models/reliers/oauth.js]: Can you validate the clientId here to ensure it's legit?

File: app/scripts/models/reliers/oauth.js [hunk 2]
```diff
@@ -87,7 +87,9 @@ var OAuthRelier = Relier.extend({
   fetch () {
     return Relier.prototype.fetch.call(this)
       .then(() => {
-        if (this._isVerificationFlow()) {
+        if (this._isSuccessFlow()) {
```
Comment (shane-tomlinson) [inline app/scripts/models/reliers/oauth.js]: This is a nice, simple solution, thanks @vladikoff!

File: app/scripts/models/reliers/oauth.js (hunk 1 above)
Comment (vladikoff) [reply app/scripts/models/reliers/oauth.js]: I Think the broker will validate it , non? or what do you mean about legit?

**Comment by vladikoff** [comment]
Added functional test

File: app/scripts/models/reliers/oauth.js (hunk 1 above)
Comment (shane-tomlinson) [reply app/scripts/models/reliers/oauth.js]: The broker won't validate it, reliers validate.

Comment (shane-tomlinson) [reply app/scripts/models/reliers/oauth.js]: And by legit, I mean, ensure the thing is a 16 character hex code.

Comment (shane-tomlinson) [reply app/scripts/models/reliers/oauth.js]: Here's what happens if garbage is put in there:

Screenshot 2019-03-26 at 10 50 54

500 error trying to decode.

Screenshot 2019-03-26 at 10 52 16

And here's what happens if the client_id is a string that doesn't match an actual client_id:

Screenshot 2019-03-26 at 10 51 05

400, better, but we could avoid the network request by validating it's a 16 character hex code.

Comment (shane-tomlinson) [reply app/scripts/models/reliers/oauth.js]: Opened #7093

**Pull request merged**